adoc/chapters/programming_interface.adoc

```diff
@@ -963,207 +963,237 @@ are handled by throwing synchronous SYCL exceptions.
 The execution environment for a SYCL application has a fixed number of platforms
 which does not vary as the application executes.
 The application can get a list of all these platforms via
-[code]#platform::get_platforms()#, and the order of the platform objects is the
+[api]#platform::get_platforms#, and the order of the platform objects is the
 same each time the application calls that function.
 The [code]#platform# class also provides constructors, but constructing a new
 [code]#platform# instance merely creates a new object that is a copy of one of
-the objects returned by [code]#platform::get_platforms()#.
+the objects returned by [api]#platform::get_platforms#.
 
-The SYCL [code]#platform# class provides the common reference semantics (see
-<<sec:reference-semantics>>).
-
-==== Platform interface
-
-A synopsis of the SYCL [code]#platform# class is provided below.
-The constructors, member functions and static member functions of the SYCL
-[code]#platform# class are listed in <<table.constructors.platform>>,
-<<table.members.platform>> and <<table.staticmembers.platform>> respectively.
-The additional common special member functions and common member functions are
-listed in <<sec:reference-semantics>> in
-<<table.specialmembers.common.reference>> and
-<<table.hiddenfriends.common.reference>> respectively.
+The [code]#platform# class provides the common reference semantics as defined in
+<<sec:reference-semantics>>.
 
-// Interface of platform class
-[source,,linenums]
+[source,role=synopsis]
 ----
 include::{header_dir}/platform.h[lines=4..-1]
 ----
 
+[[sec:platform-ctors]]
+==== Constructors
 
-[[table.constructors.platform]]
-.Constructors of the SYCL [code]#platform# class
-// Jon: Dims{6.7cm}{8.3cm}
-[width="100%",options="header",separator="@",cols="65%,35%"]
-|====
-@ Constructor @ Description
-a@
-[source]
+.[apititle]#Default constructor#
+[source,role=synopsis,id=api:platform-ctor]
 ----
 platform()
 ----
-   a@ Constructs a SYCL [code]#platform# instance that is a copy of the
-      [code]#platform# which contains the device returned by
-      [code]#default_selector_v#.
 
-a@
-[source]
+_Effects:_ Constructs a [code]#platform# object that is a copy of the platform
+which contains the device returned by [code]#default_selector_v#.
+
+'''
+
+.[apititle]#Selector constructor#
+[source,role=synopsis,id=api:platform-ctor-selector]
 ----
-template <typename DeviceSelector> explicit platform(const DeviceSelector&)
+template <typename DeviceSelector>
+explicit platform(const DeviceSelector& selector)
 ----
-   a@ Constructs a SYCL [code]#platform# instance that is a copy of the
-      [code]#platform# which contains the device returned by the
-      <<device-selector>> parameter.
 
-|====
+_Constraints:_ The [code]#DeviceSelector# must be a type that satisfies the
+requirements of a <<device-selector>> as defined in <<sec:device-selector>>.
 
+_Effects:_ The [code]#selector# is called once for every <<root-device>> as
```

Comment:
Remove the `once`? Who knows, maybe some implementation wants to do two passes on the list of rood devices?

Reply:
I did this in 1b1bde486d535fe9d211b1c82d05ec6e4b5fe80a because I agree that [this section](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:device-selection) does not currently say that the selector is called exactly once for each device.

However, I wonder if this is an oversight in the spec.  Why wouldn't we want to guarantee that it is called exactly once for each root device?  This could make a difference for a selector that maintains internal state across each  call.

Reply:
I agree, but that seems more than editorial as, in theory, it may impact a state-full selector, as you said. 

I don't have a strong opinion about this one, but I lean toward adding `once` :) 

It doesn't seem like a big ask for an implementer to call their selector only once per device, and indeed, it may simplify the user's implementation of a selector

Reply:
FWIW, I think these "new format" PRs will be more than pure editorial, so it should be OK to make some small improvements.  I don't mind making this as a separate PR, though, if you prefer.

Reply:
Maybe a new PR would give others an opportunity to chime in, and we could merge this PR without waiting for the "once" resolution.